packages/commerce-sdk-react/src/hooks/ShopperBaskets/query.ts

```diff
@@ -6,73 +6,219 @@
  */
 import {ApiClients, Argument, DataType} from '../types'
 import {useQuery} from '../useQuery'
-import useCommerceApi from '../useCommerceApi'
-import {UseQueryResult} from '@tanstack/react-query'
+import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
 
 type Client = ApiClients['shopperBaskets']
 
+type UseBasketParameters = NonNullable<Argument<Client['getBasket']>>['parameters']
+type UseBasketHeaders = NonNullable<Argument<Client['getBasket']>>['headers']
+type UseBasketArg = {
+    headers?: UseBasketHeaders
+    rawResponse?: boolean
+} & UseBasketParameters
 /**
  * A hook for `ShopperBaskets#getBasket`.
  * Gets a basket.
  * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=getBasket} for more information about the API endpoint.
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperbaskets.shopperbaskets-1.html#getbasket} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const useBasket = (
-    arg: Argument<Client['getBasket']>
-): UseQueryResult<DataType<Client['getBasket']>, Error> => {
-    const {shopperBaskets: client} = useCommerceApi()
-    return useQuery(['baskets', arg], () => client.getBasket(arg))
+function useBasket(
+    arg: Omit<UseBasketArg, 'rawResponse'> & {rawResponse?: false},
+    options?: UseQueryOptions<DataType<Client['getBasket']> | Response, Error>
+): UseQueryResult<DataType<Client['getBasket']>, Error>
+function useBasket(
+    arg: Omit<UseBasketArg, 'rawResponse'> & {rawResponse: true},
+    options?: UseQueryOptions<DataType<Client['getBasket']> | Response, Error>
+): UseQueryResult<Response, Error>
+function useBasket(
+    arg: UseBasketArg,
+    options?: UseQueryOptions<DataType<Client['getBasket']> | Response, Error>
+): UseQueryResult<DataType<Client['getBasket']> | Response, Error> {
+    const {headers, rawResponse, ...parameters} = arg
+    const defaultOptions = {
+        enabled: !!parameters.basketId
+    }
```

Comment:
by default, the queries should be disabled, if the IDs of the entity doesn't exist.

Reply:
@kevinxh  let's pull this out from the PR and make another PR to have this change implemented globally.